### DISCUSSION TITLE
File Types

### DISCUSSION THREAD

**Discussion by Rycha2021** [iTwin/itwinjs-core#2885] [Q&A]
@pmconne, in a post https://github.com/iTwin/itwinjs-core/discussions/2253#discussioncomment-1333786, your response is somewhat confusing, at least to me.

You first say: 
"_Yes. Both 3mx and 3sm support publishing as 3d tiles (sometimes referred to as "web-ready tiles"). iModel.js supports all 3d tile formats and it doesn't care what the original reality model format was._"
Then you next say:
"_iModel.js **does not support displaying** .3mx or .3sm files directly_."

What is your definition of "publishing" and "displaying"?

We would prefer to use 3mx files within iTwin for various reasons. If possible we will also look to use classifiers etc. 

Can iTwin use and display 3mx files as easily as 3sm files or have i been advised incorrectly and have likely burnt hours developing on the wrong platform?

In the past i used an old version of context capture viewer using 3mx files and found this gave good visual results. Can we expect the same quality / results in iTwin as per context capture viewer?

Thanks

Ry

**Comment by pmconne** [comment]
3d tiles are an [open-source, standardized specification](https://www.ogc.org/standards/3DTiles) optimized for efficient streaming of 3d content over the web, based on [glTF](https://www.khronos.org/registry/glTF/), the open-source standardized specification for defining 3d models for the web (aka "the JPEG of 3d").
3mx and 3sm are [non-standardized formats](https://docs.bentley.com/LiveContent/web/ContextCapture%20Help-v18/en/GUID-817E1113-2882-46CF-8C07-549ECCA04BCE.html) maintained by Bentley Systems describing multi-resolution textured meshes. Both can be easily converted (aka "published") to 3d tiles. For example, [ContextCapture WebViewer 2.0](https://docs.bentley.com/LiveContent/web/ContextCapture%20Help-v18/en/GUID-0CD08E66-DD9A-4448-8FCD-9EF6A410AD45.html) converts 3mx to 3d tiles for web viewing.
iTwin.js displays 3d tiles. Those 3d tiles may have been converted from 3sm, 3mx, or any other format. That's the point of having open standardized formats - viewing applications only need to support the standard, and other formats need only support conversion to the standard.
You should expect the same quality in iTwin using 3d tiles as in the old ContextCapture viewer using 3mx files (better, in fact, given [this caveat about streaming city-scale 3mx models over the web](https://docs.bentley.com/LiveContent/web/ContextCapture%20Help-v18/en/GUID-FEED2A71-66F5-4042-95E8-0DA2D827E21D.html)).

**Comment by Rycha2021** [comment]
Thank-you for the timely response @pmconne 

Are there any quirks / issues / limitations with using 3MX in iTwin? Does all the other functionality possible work, e.g. classifiers and other API's etc?

I note in the link you refer to says:

_Note: City-scale 3D models are split into tiles when using ContextCapture , and the level-of-detail (LOD) structure used by the 3MX format is generated for each tile independently. **This leads to an incomplete level-of-detail structure and low loading performance if the model contains a large number of tiles**. As this is especially visible in a web environment, **BentleySystems has developed a post-processing tool capable of creating the entire LOD structure. This tool, called ExportUniqueMesh, is currently in its beta version**._

Is this old so 'Beta' version is now available?

Thanks. Ry

**Reply by rbois-bentley** [reply]
@Rycha2021 3MX isn't supported in iTwin but you can convert them to 3dtiles for viewing in iTwin. Classifiers are also supported but I'm not sure what you mean by "other" APIs?

Concerning ExportUniqueMesh, it should be available in latest release of ContextCapture.

**Reply by ovaisa** [reply]
Can you please advise how I can convert 3mx to 3d Tile and publish to ContextShare for streaming into iTwin app.  I am using Descartes (which seems to support only 3sm), is there an option in ContextCapture to publish web ready tiles using 3mx to ContextShare.

**Reply by ovaisa** [reply]
In some cases we have our models in Cesium 3D tiles format, it would be lot better if we can publish 3d tiles/Web Ready tiles to ContextShare (to avoid any loss of quality in the conversion process).  Can you please advise how we can do that.

**Reply by Pascal-Cloutier** [reply]
You can convert a 3MX or 3SM to Web Ready Scalable Mesh (Bentley's flavor of 3D Tiles) using the _Orbit 3DM Import And Upload Tool_. The same app will also allow you to upload existing models in Cesium 3D Tiles format to ContextShare. Here is a link to a wiki page that offers more details on the workflow: [Upload reality mesh from Orbit 3DM Import And Upload Tool to ProjectWise ContextShare](https://communities.bentley.com/products/3d_imaging_and_point_cloud_software/w/wiki/55907/upload-reality-mesh-from-orbit-3dm-import-and-upload-tool-to-projectwise-contextshare)

Note that you also have the option of uploading your existing models in Cesium 3D Tiles format using the [Reality Data API](https://developer.bentley.com/apis/reality-data/).
